### PR TITLE
Adding releasing process guide

### PR DESCRIPTION
**Pull request by srprash**

### CODE REVIEW

**Comment by srprash** [comment]
Will fix the linting issues shortly.

**Comment by srprash** [comment]
@cijothomas Can you take a look?

File: build/RELEASING.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    ```
+
+    It usually takes a while for the new version of the project to show up in nuget.org
+
+7. Validate that the new version of the project is successfully published to
+nuget.org under OpenTelemetry owner.
+
+8. Delete the NUGET TOKEN set in step 6.
+
+9. Go to the [**Releases**](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/releases)
```
Comment (cijothomas) [inline build/RELEASING.md:83]: Could you add putting changelog from the released component in the github release as well.
Also, this should be a final one, *not* draft :)

File: build/RELEASING.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+## Pre-step
+
+Before starting with the release process, ensure that the latest commit on
+the `main` branch is the commit which has added/updated the Changelog to
+the project being released. This is the commit that will be tagged on the
+release.
+
+## Steps
+
+1. Create and push git tag for the project and the version of the project
```
Comment (cijothomas) [inline build/RELEASING.md:27]: please add atleast one example, like
git tag -a 1.0.0-rc2 -m "1.0.0-rc2"
git push origin 1.0.0-rc2

File: build/RELEASING.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+you want to release.
+
+    ```powershell
+    git tag -a <PROJECT TAG PREFIX><VERSION> -m "<PROJECT TAG PREFIX><VERSION>"
+    ```
+
+    You can find the project tag prefix in the workflow file for the project.
```
Comment (cijothomas) [inline build/RELEASING.md:34]: the source of truth should be the .csproj. example : https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Contrib.Instrumentation.Elasticsearch/OpenTelemetry.Contrib.Instrumentation.ElasticsearchClient.csproj#L6

File: build/RELEASING.md (hunk 1 above)
Comment (cijothomas) [reply build/RELEASING.md:83]: Just realized we are missing a pre-step  - which is to update changelog to change "unreleased" to actual version and date of release. This must be a pre-step. (similar to the main repo)

**Review by cijothomas — CHANGES_REQUESTED** [review]
Thanks!

Tried the steps for MT and ES released.
Shared feedback about what needs changes from the above attempts.

**Comment by srprash** [comment]
Nice! Thanks for the pointers. Will add those in the guide. :)

File: build/RELEASING.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+`main` branch and create a GitHub release later on.
+
+2. Have access to the Nuget token for releasing the project to Nuget.org
+OpenTelemetry account.
+
+## Pre-step
+
+Before starting with the release process, ensure that the latest commit on
+the `main` branch is the commit which has added/updated the Changelog to
+the project being released. The Changelog at this point **must** include the
```
[the remark this answers is not in the document]
Comment (cijothomas) [reply build/RELEASING.md:22]: yes. (powershell or manual step)

File: build/RELEASING.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+1. You must be a maintainer on the repo. This will allow you to push tags to
+`main` branch and create a GitHub release later on.
+
+2. Have access to the Nuget token for releasing the project to Nuget.org
+OpenTelemetry account.
+
+## Pre-steps
+
+1. Update the Changelog for your project with relevant details. You
```
Comment (cijothomas) [inline build/RELEASING.md:20]: How about:
1. Decide on the version to be used.
2. Update changelog to replace Unreleased with this version and add date of release.
3. Submit PR to update changelog and get it merged.

and in the Steps.1 - mention that the version should be same as the one updated in changelog.

**Pull request merged**